pkgs/development/tools/fermyon_spin/default.nix

```diff
@@ -0,0 +1,49 @@
+{ lib
+, stdenv
+, fetchzip
+, autoPatchelfHook
+, zlib
+, gcc-unwrapped
+}:
+
+stdenv.mkDerivation rec {
```

Comment:
Since https://github.com/fermyon/spin/commit/445906dde996e3d6e91439d80a22645b78c51c20 we should be able to build from source in nix rather than relying on binary releases - IIRC I had a WIP branch somewhere but hadn't quite figured out how to make vergen happy

Reply:
Here's the initial source version for the reference.
I got stuck at the `outputHashes`, as the `wit-bindgen-gen-core` is specified twice and I cannot figure out how to add a second hash for the same entry.
<details>
<summary>Initial source version</summary>

```nix
{ lib
, stdenv
, rustPlatform
, fetchFromGitHub 
}:

rustPlatform.buildRustPackage rec {
  pname = "fermyon_spin";
  version = "1.1.0";

  src = fetchFromGitHub {
    owner = "fermyon";
    repo = "spin";
    rev = "v${version}";
    sha256 = "sha256-WauKRih/X4r43t4bAsfCcEWkYDwvOh0lRwGimayjCVw=";
  };

  cargoLock = {
    lockFile = "${src}/Cargo.lock";
    outputHashes = {
      "bindle-0.8.0" = lib.fakeHash;
      "cargo-target-dep-0.1.0" = lib.fakeHash;
      "cloud-openapi-0.1.0" = lib.fakeHash;
      "dkregistry-0.5.1-alpha.0" = lib.fakeHash;
      "hippo-0.16.1" = lib.fakeHash;
      "host-0.0.0" = lib.fakeHash;
      "oci-distribution-0.9.3" = lib.fakeHash;
      "wasm-encoder-0.25.0" = lib.fakeHash;
      "wit-bindgen-gen-core-0.2.0" = lib.fakeHash;
    };
  };

  meta = with lib; {
    description = "Framework for building, deploying, and running fast, secure, and composable cloud microservices with WebAssembly.";
    homepage = "https://github.com/fermyon/spin";
    license = with licenses; [ asl20 ];
    maintainers = with maintainers; [ mglolenstine ];
    platforms = platforms.linux;
  };
}
  ```
</details>

Reply:
Whats the actual error? If it can't be solved within nix because they have the same version, you probably need to patch the cargo file.

Reply:
<details>
<summary>This is the exact error</summary>

```
building '/nix/store/87s3j1s886xm34mb0gm3anavw83r4fc0-source.drv'...

trying https://github.com/fermyon/spin/archive/v1.1.0.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 5565k    0 5565k    0     0   881k      0 --:--:--  0:00:06 --:--:--  947k
unpacking source archive /build/v1.1.0.tar.gz
error: No hash was found while vendoring the git dependency wit-bindgen-gen-core-0.2.0. You can add
       a hash through the `outputHashes` argument of `importCargoLock`:

       outputHashes = {
         "wit-bindgen-gen-core-0.2.0" = "<hash>";
       };

       If you use `buildRustPackage`, you can add this attribute to the `cargoLock`
       attribute set.
(use '--show-trace' to show detailed location information)
```
</details>

Reply:
I realised that `v1.2.1` has been released, and while some dependencies do change, the source still doesn't want to build due to the above error.
(Updating it to `v1.2.1`)